### PR TITLE
socketlin : add lin.h and lin bus-type

### PR DESCRIPTION
**Pull request by HedongGao**
Add lin.h adn lin bus-type

## Summary
Add lin.h for socket lin.

## Impact
New feature

## Testing
Write LIN master and slave programs, flash them to two development boards respectively, and test LIN transceiver business.

### CODE REVIEW

File: include/nuttx/lin.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#ifndef __INCLUDE_NUTTX_LIN_H
+#define __INCLUDE_NUTTX_LIN_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/can.h>
+#include <netpacket/can.h>
```
Comment (xiaoxiang781216) [inline include/nuttx/lin.h:30]: remove, doesn't exist anymore

Comment (HedongGao) [reply include/nuttx/lin.h]: Done！

**Review by acassis — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@pkarashchenko could you take a look?

**Comment by pkarashchenko** [comment]
Yes. Will take a look today

File: include/nuttx/lin.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#define __INCLUDE_NUTTX_LIN_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/can.h>
+
+#ifdef CONFIG_NET_CAN
```
Comment (pkarashchenko) [inline include/nuttx/lin.h:31]: Looks good, but one question here. Why do we need this ifdef wrapper? Is "classic" LIN (same as CAN) driver going to be supported in parallel?

**Review by pkarashchenko — APPROVED** [review]

File: include/nuttx/lin.h (hunk 2 above)
Comment (HedongGao) [reply include/nuttx/lin.h]: This is indeed a problem, and I have corrected it！

**Pull request merged**